### PR TITLE
Add GitHub Actions to build on Windows, Linux and Mac

### LINKED ISSUE #3812: Add GitHub Actions
I'd like to see us using GitHub Actions to build this project so that everything is in one place. Once we do that we can make a decision on which pipelines we will maintain going forward and which will be our primary pipeline.

@OsirisTerje linked this project that runs GitHub Actions locally for testing, https://github.com/nektos/act

### PR DESCRIPTION
**Pull request by rprouse**
Fixes #3812

Currently just builds on all three platforms and saves the packages from Windows. It does not publish the packages to any NuGet package feed. In the future, we may do that.

### CODE REVIEW

**Comment by rprouse** [comment]
This is looking good now. Marking ready for review and will make Required once it is merged.

File: .github/workflows/continuous_integration.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+jobs:
+  build-windows:
+    runs-on: windows-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Build and Test
+      run: .\build.ps1 --target=Test --test-run-name=Windows
```
Comment (manfred-brands) [inline .github/workflows/continuous_integration.yml:23]: Any reason why you not use 'dotnet cake' to match the other two OS?

File: .github/workflows/continuous_integration.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  push:
+    branches:
+      - master
+      - release
+      - 'v3.13-dev'
+  pull_request:
+    branches:
+      - master
+      - release
+      - 'v3.13-dev'
```
Comment (jnm2) [inline .github/workflows/continuous_integration.yml:13]: Is there any reason to block CI from running if there was ever a pull request to another branch?
There might be other better-looking ways to do it, but `pull_request:` by itself with nothing following it enables the pull request trigger without putting filters on it. Maybe `branches: [ '*' ]` underneath if that looks too weird.

**Review by jnm2 — APPROVED** [review]

**Comment by jnm2** [comment]
One thing I'd like to do eventually is add binlog artifacts. That makes it easy to drop into https://msbuildlog.com if you have it installed to diagnose weird CI issues if they come up. It captures all environment variables though, so it doesn't work if there are decrypted secrets in there.

Here's what it looks like to create binlogs (three of them in this case): https://github.com/shouldly/shouldly/blob/master/build.ps1#L15-L29

And upload them: https://github.com/shouldly/shouldly/blob/037a2b9c118aa266a84988d9adf0301c4ca1ad5a/.github/workflows/CI.yml#L29-L34

File: .github/workflows/continuous_integration.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        name: Package
+        path: package
+
+    - name: Upload test results
+      uses: actions/upload-artifact@v2
+      with:
+        name: Test results (Windows)
+        path: test-results
+      # Use always() to always run this step to publish test results when there are test failures
+      if: ${{ always() }}
```
Comment (jnm2) [inline .github/workflows/continuous_integration.yml:40]: `if: always()` has been working for me.

Looks like the `${{` escape is only needed if you aren't already implicitly in an expression context or if you have operators: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#about-contexts-and-expressions

**Comment by rprouse** [comment]
@manfred-brands thanks for the review. I apologize, I mis-tagged you, but I appreciate your help 👍🏻

**Comment by jnm2** [comment]
Just thought of one more thing to recommend:

```yml
env:
  DOTNET_NOLOGO: true
```

File: .github/workflows/continuous_integration.yml (hunk 3 above)
Comment (rprouse) [reply .github/workflows/continuous_integration.yml:40]: I took this directly from the GitHub docs, but good to know.

File: .github/workflows/continuous_integration.yml (hunk 2 above)
Comment (rprouse) [reply .github/workflows/continuous_integration.yml:13]: You are probably right. I might change this in my next PR.

**Comment by rprouse** [comment]
@manfred-brands thanks for the review. I apologize, I mis-tagged you

File: .github/workflows/continuous_integration.yml (hunk 2 above)
Comment (jnm2) [reply .github/workflows/continuous_integration.yml:13]: I wonder if `pull_request` with no `:` would have worked.

Comment (jnm2) [reply .github/workflows/continuous_integration.yml:13]: > if there was ever a pull request to another branch?

It sounds unusual, but I've contributed directly to the topic branches of another person's PRs on GitHub. PR inception. I've also contributed to a compiler feature branch which is not something NUnit currently does.

**Pull request merged**